api/types/resource.go

```diff
@@ -277,3 +281,36 @@ var validLabelKey = regexp.MustCompile(LabelPattern)
 func IsValidLabelKey(s string) bool {
 	return validLabelKey.MatchString(s)
 }
+
+// MatchSearch goes through select field values from a resource
+// and tries to match against the list of search values, ignoring case and order.
+// Returns true if all search vals were matched (or if nil search vals).
+// Returns false if no or partial match (or nil field values).
+func MatchSearch(fieldVals []string, searchVals []string, customMatch func(val string) bool) bool {
+	// Case fold all values to avoid repeated case folding while matching.
+	for i, val := range fieldVals {
+		fieldVals[i] = strings.ToLower(val)
+	}
+	for i, val := range searchVals {
+		searchVals[i] = strings.ToLower(val)
+	}
```

Comment:
This modifies the slices in place, which would be quite unexpected for the caller - we pass `searchVals` directly from the `values` parameter of `(ResourceWithLabels).MatchSearch` in all cases, AFAICT. Either use `utils.CopyStrings` or make a new utility function that lowers a slice of strings at once, I guess?